nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/elasticsearch/ElasticsearchObject.java

```diff
@@ -64,7 +65,51 @@ public void doIndexing(@PathParam("repositoryName") String repositoryName) {
         });
     }
 
+    /**
+     * Executes an ES flush on all documents indexes of a given repository.
+     */
+    @POST
+    @Path("{repositoryName}/flush")
+    public void doFlush(@PathParam("repositoryName") String repositoryName) {
+        Repository repository = getRepository(repositoryName);
+        Framework.doPrivileged(() -> {
+            Framework.getService(ElasticSearchAdmin.class).flushRepositoryIndex(repository.getName());
+        });
+    }
+
+    /**
+     * Executes an ES optimize on all documents indexes of a given repository.
```

Comment:
on the document index of a given repository (there's only one document index peer repository)

Reply:
i think i missed something, a repository will hold one index but many documents. for example we will have an User index, and many users (documents) will be indexed. it's my understanding

Reply:
Afaik there's no user index in ES, we index the repository (one index per repository), the audit and a sequence.

In case of re-indexing, we may have a second index for the repository, representing the current indexing. Which allows us to not break service during indexation and just change the alias of the index once done.